### PR TITLE
[16.0][MIG] sale_invoicing_date_selection: Backport from 17.0

### PR DESCRIPTION
**Pull request by sergiobstoj**
Backport from: https://github.com/OCA/account-invoicing/pull/1767

### CODE REVIEW

File: sale_invoicing_date_selection/tests/test_sale_invoicing_date_selection.py [hunk 1]
```diff
@@ -48,10 +48,9 @@ def test_andvance_invoice(self):
             )
         ) as wiz_invoice_form:
             wiz_invoice_form.invoice_date = "2024-03-01"
+            wiz_invoice_form.advance_payment_method = "fixed"
+            wiz_invoice_form.fixed_amount = 10
         wiz = wiz_invoice_form.save()
-        # set values by code as there are invisible fields
```
Comment (aritzolea) [inline sale_invoicing_date_selection/tests/test_sale_invoicing_date_selection.py]: I'd say visibility conditions did not change from V16 to V17, why change this?

**Review by aritzolea — CHANGES_REQUESTED** [review]
Separate please pre-commit commit from the actual migration commit as done always in migrations please.

File: sale_invoicing_date_selection/tests/test_sale_invoicing_date_selection.py (hunk 1 above)
Comment (sergiobstoj) [reply sale_invoicing_date_selection/tests/test_sale_invoicing_date_selection.py]: When using just one `sale.order` record, as in this case, those fields are not invisible.
Also, when having:
```
# set values by code as there are invisible fields
wiz.advance_payment_method = "fixed"
wiz.fixed_amount = 10
```
it returns an error during testing:
```
raise UserError(_('The value of the down payment amount must be positive.'))
odoo.exceptions.UserError: El valor del importe por adelantado debe ser positivo.
```
Even trying `save()` after setting those values
So it wasn't making any effect.

**Review by aritzolea — APPROVED** [review]

**Review by PabloMartFL — APPROVED** [review]
Functional Review LGTM!

Whenever you create an invoice by the action menu and select any date, the invoice will be dated on the selected date

https://www.awesomescreenshot.com/video/31218344?key=2dbfa58742677505d5e6a79ea254b154

**Comment by sergiobstoj** [comment]
@sergio-teruel I understand you are maintainer of this module in 17.0. Is this OK for you to merge?

**Review by sergio-teruel — APPROVED** [review]

**Comment by sergio-teruel** [comment]
@sergiobstoj Thanks..

**Review by pedrobaeza — APPROVED** [review]
/ocabot migration sale_invoicing_date_selection
/ocabot merge nobump

**Pull request merged**